x-pack/plugins/maps/public/actions/store_actions.js

```diff
@@ -165,7 +168,11 @@ export function toggleLayerVisible(layerId) {
 
 export function setSelectedLayer(layerId) {
   return async (dispatch, getState) => {
-
+    const transientLayerId = getTransientLayerId(getState());
```

Comment:
would not wrap transientLayerState management inside setSelectedLayer. Those two are really unrelated (see related longer comment). The reason this is wrapped up in here is because you're calling `setSelectedLayer` inside the `addTransientLayer` action-creator. It's a little confusing because the add_transient_layer action creator does a lot more than just dispatching the `add_transient_layer` action. (hence also the suggestion to just rename that action)

Reply:
It's necessary to check & remove any transient layers on `setSelectedLayer` to prevent the case when a user starts adding a new layer but then switches to looking at another layer's properties. In our current workflow, a transient layer is always a selected layer so I think the call should stay. I'm open to renaming `addTransientLayer`, maybe `addTransientLayerAndSetSelected`?

Reply:
I'd keep all this new transient-layer code parallel from our existing action-creators like setSelectedLayer. 90% of cases, setting layer selection has nothing to do with transient-layer-state. Transient-layer state is a very fringe-concept in the app, very shortlived. 

For this particular case (which I missed in the list of 4 above), we know exactly how it can arise. A user clicks on a layer-entry, while the add-source-wizard editor is open. You could handle that case when that even occurs, before setting the layer-selection.

e.g.:
https://github.com/thomasneirynck/kibana/blob/de130211d8c08bbd0bdb6d72d649d42461f7809b/x-pack/plugins/maps/public/components/widget_overlay/layer_control/layer_toc/toc_entry/index.js#L30

e.g. before this line, check if there's a transient layer. If so, use removeLayer to remove it from the layer-list and then just have all the rest execute.

Reply:
I ended up leaving this in for a couple of reasons. To check the selector at the point you mentioned, I'd need to provide an arg of state, which I don't have access to in the container. There are ways around this like using the selector in `mapStateToProps`, getting the id and doing a roundtrip through the component back to the container, but that's not ideal. I do agree we're primarily just using this via the TOC entries, but I could also see us selecting layers in other ways moving forward via map click, fitToBounds, etc.